src/Common/levenshteinDistance.h

```diff
@@ -1,12 +1,112 @@
 #pragma once
 
 #include <base/types.h>
+#include <Common/PODArray.h>
+#include <Common/iota.h>
+#include <span>
+
+#include <numeric>
 
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int SIZES_OF_ARRAYS_DONT_MATCH;
+}
+
 /// How many steps if we want to change lhs to rhs.
 /// Details in https://en.wikipedia.org/wiki/Levenshtein_distance
 size_t levenshteinDistance(const String & lhs, const String & rhs);
```

Comment:
Let's remove the old variant and the cpp file, because `levenshteinDistance<char>` will work for strings.

Reply:
Unfortunately, it didn't work out https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=77187&sha=2eff9d005643d1c0570ffbde6388455782c1664c&name_0=PR&name_1=Fast+test&name_2=Tests

The old function is case insensitive, and I don't think it makes sense to make another template for it.

I'll bring it back and rename to `levenshteinDistanceCaseInsensitive`